pkg/river/ast/ast.go

```diff
@@ -0,0 +1,309 @@
+// Package ast exposes AST elements used by River.
+//
+// The various interfaces exposed by ast are all closed; only types within this
+// package can satisfy an AST interface.
+package ast
+
+import (
+	"fmt"
+
+	"github.com/grafana/agent/pkg/river/token"
+)
+
+// Node represents any node in the AST.
+type Node interface {
+	astNode()
+}
+
+// Stmt is a type of statement wthin the body of a file or block.
+type Stmt interface {
+	Node
+	astStmt()
+}
+
+// Expr is an expression within the AST.
+type Expr interface {
+	Node
+	astExpr()
+}
+
+// File is a parsed file.
+type File struct {
```

Comment:
I was thinking of the scraping service, if it loads the configs as sub-vms/asts.

Reply:
What about Config?

Reply:
Thinking about it again, File or Source is probably preferable over Config, since Config implies it's a complete config, which may not be true in the future when we load multiple files.

Reply:
That’s reasonable